pkg/util/util.go

```diff
@@ -89,3 +89,15 @@ func ReverseMap[K comparable, V comparable](m map[K]V) map[V]K {
 
 	return new
 }
+
+// FilterMap filters a map, only keeping the key-value pairs that pass the keep function
```

Comment:
💡minor: might be add a sample e.g. below this comment.

Reply:
they're failing due to quota

https://github.com/Azure/aks-app-routing-operator/actions/runs/9225150664

at some point we should make our e2e choose region more dynamically based on quota